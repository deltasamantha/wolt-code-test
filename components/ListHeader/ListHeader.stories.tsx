import {Meta, StoryObj} from "@storybook/react-native";
import {View} from "react-native";

import ListHeader, {Props} from "./ListHeader";
import {storyStyles} from "./ListHeader.styles";

const ListHeaderMeta: Meta<Props> = {
  title: "ListHeader",
  component: ListHeader,
  decorators: [
    Story => (
      <View style={storyStyles.rootContainer}>
        <Story />
      </View>
    ),
  ],
};

export default ListHeaderMeta;

export const Primary: StoryObj<Props> = {
  args: {
    title: "Opening hours",
  },
};
