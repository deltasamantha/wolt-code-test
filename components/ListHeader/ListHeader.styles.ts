import {StyleSheet} from "react-native";

import theme from "../../theme/theme";

export const styles = StyleSheet.create({
  rootContainer: {
    flexDirection: "column",
    backgroundColor: theme.colors.colorWhite,
    paddingTop: theme.spacing.lg,
    borderTopStartRadius: theme.borderRadius.md,
    borderTopEndRadius: theme.borderRadius.md,
    paddingHorizontal: theme.spacing.lg,
  },
  titleContainer: {
    flexDirection: "row",
    alignItems: "center",
    paddingBottom: theme.spacing.sm,
  },
  title: {
    marginLeft: theme.spacing.md,
  },
  clock: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: theme.colors.colorLight,
  },
});

export const storyStyles = StyleSheet.create({
  rootContainer: {
    flex: 1,
    paddingHorizontal: theme.spacing.lg,
    backgroundColor: theme.colors.colorBackground,
    justifyContent: "center",
  },
});
