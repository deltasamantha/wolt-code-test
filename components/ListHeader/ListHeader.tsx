import React from "react";
import {View} from "react-native";

import {styles} from "./ListHeader.styles";
import AppText from "../AppText/AppText";
import Divider from "../Divider/Divider";

interface Props {
  title: string;
}

const ListHeader: React.FC<Props> = ({title}: Props) => {
  return (
    <View style={styles.rootContainer}>
      <View style={styles.titleContainer}>
        <View style={styles.clock} />
        <AppText size="l" type="bold" style={styles.title}>
          {title}
        </AppText>
      </View>
      <Divider />
      <View />
    </View>
  );
};

export default ListHeader;
