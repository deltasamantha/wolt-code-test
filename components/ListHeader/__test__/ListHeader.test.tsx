import {render, cleanup, screen} from "@testing-library/react-native";

import locale from "../../../config/Locale";
import ListHeader from "../ListHeader";

afterEach(cleanup);

describe("ListHeader", () => {
  it("Should render ListHeader properly", () => {
    const {toJSON} = render(<ListHeader />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("ListHeader should render title", () => {
    const sampleText = locale.slotsListTitle;
    render(<ListHeader />);
    const text = screen.getByText(sampleText);
    expect(text.props.children).toBeDefined();
    expect(text.props.children).toEqual(sampleText);
  });

  it("ListHeader should have divider", () => {
    render(<ListHeader />);
    const divider = screen.getByTestId("divider");
    expect(divider).toBeTruthy();
  });

  it("ListHeader should have clock image", () => {
    render(<ListHeader />);
    const divider = screen.getByTestId("clockImage");
    expect(divider).toBeTruthy();
  });
});
