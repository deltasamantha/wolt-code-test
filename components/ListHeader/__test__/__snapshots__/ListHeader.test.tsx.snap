// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListHeader Should render ListHeader properly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderTopEndRadius": 10,
      "borderTopStartRadius": 10,
      "flexDirection": "column",
      "paddingHorizontal": 16,
      "paddingTop": 16,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 10,
      }
    }
  >
    <Image
      source={1}
      style={
        {
          "height": 30,
          "width": 30,
        }
      }
      testID="clockImage"
    />
    <Text
      style={
        [
          {
            "color": "#202125",
            "fontFamily": "Roboto-Bold",
            "fontSize": 24,
            "lineHeight": 30,
          },
          {
            "marginLeft": 10,
          },
        ]
      }
    >
      Opening hours
    </Text>
  </View>
  <View
    style={
      [
        {
          "height": 1,
          "width": "100%",
        },
        {
          "backgroundColor": "#202125",
        },
      ]
    }
    testID="divider"
  />
</View>
`;
